evalml/tests/component_tests/test_oversamplers.py

```diff
@@ -0,0 +1,244 @@
+        if 'NC' in sampler.name:
+            X = infer_feature_types(X, feature_types={1: "Categorical"})
+            oversampler = sampler(sampling_ratio=1, random_seed=seed)
+        samplers.append(oversampler)
+
+    # iterate through different indices in samplers
+    # in group 1, first two oversamplers in samplers should be equal
+    # in group 2, calling same oversamplers twice should be equal
+    # in group 3, last two oversamplers in samplers should be different
+    for s1, s2 in [[0, 1], [1, 1], [1, 2]]:
+        X1, y1 = samplers[s1].fit_transform(X, y)
+        X2, y2 = samplers[s2].fit_transform(X, y)
+        if s2 == 2 and sampler != SMOTENSampler:
+            # group 3, SMOTENSampler performance doesn't change with different random states
+            with pytest.raises(AssertionError):
+                pd.testing.assert_frame_equal(X1.to_dataframe(), X2.to_dataframe())
+        else:
+            pd.testing.assert_frame_equal(X1.to_dataframe(), X2.to_dataframe())
+        pd.testing.assert_series_equal(y1.to_series(), y2.to_series())
+
+
+@pytest.mark.parametrize("component_sampler,imblearn_sampler",
+                         [(SMOTESampler, im.SMOTE),
+                          (SMOTENCSampler, im.SMOTENC),
+                          (SMOTENSampler, im.SMOTEN)])
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
+def test_samplers_perform_equally(problem_type, component_sampler, imblearn_sampler, X_y_binary, X_y_multi):
+    if problem_type == 'binary':
+        X, _ = X_y_binary
+        y = np.array([0] * 90 + [1] * 10)
+        imb_learn_sampling_ratio = 0.5
+        expected_y = np.array([0] * 90 + [1] * 45)
+    else:
+        X, _ = X_y_multi
+        y = np.array([0] * 70 + [1] * 20 + [2] * 10)
+        imb_learn_sampling_ratio = {0: 70, 1: 35, 2: 35}
+        expected_y = np.array([0] * 70 + [1] * 35 + [2] * 35)
+    sampling_ratio = 0.5
+    sampling_dic = {'sampling_ratio': sampling_ratio}
+    X2 = X
+    random_seed = 1
+    if component_sampler != SMOTENCSampler:
+        component = component_sampler(**sampling_dic, random_seed=random_seed)
+        imb_sampler = imblearn_sampler(sampling_strategy=imb_learn_sampling_ratio, random_state=random_seed)
+    else:
+        X2 = infer_feature_types(X, feature_types={1: "Categorical", 2: "Categorical", 3: "Categorical", 4: "Categorical"})
+        component = component_sampler(**sampling_dic, random_seed=random_seed)
+        imb_sampler = imblearn_sampler(sampling_strategy=imb_learn_sampling_ratio, categorical_features=[1, 2, 3, 4], random_state=random_seed)
+
+    X_com, y_com = component.fit_transform(X2, y)
+    X_im, y_im = imb_sampler.fit_resample(X, y)
+
+    np.testing.assert_equal(X_com.to_dataframe().values, X_im)
+    np.testing.assert_equal(y_com.to_series().values, y_im)
+    np.testing.assert_equal(sorted(y_im), expected_y)
+
+
+def test_smotenc_no_categorical_features_passed(X_y_binary):
```

Comment:
nit-pick: This test name is misleading cause we're passing categorical features?

Reply:
Good call!